modyn/supervisor/internal/grpc_handler.py

```diff
@@ -434,3 +462,173 @@ def seed_selector(self, seed: int) -> None:
         ).success
 
         assert success, "Something went wrong while seeding the selector"
+
+    def start_evaluation(self, trained_model_id: int, pipeline_config: dict) -> dict[int, EvaluationStatusTracker]:
+        if not self.connected_to_evaluator:
+            raise ConnectionError("Tried to start evaluation at evaluator, but there is no gRPC connection.")
+
+        model_id = pipeline_config["model"]["id"]
+        if "config" in pipeline_config["model"]:
+            model_config = json.dumps(pipeline_config["model"]["config"])
+        else:
+            model_config = "{}"
+
+        device = pipeline_config["evaluation"]["device"]
+        amp = pipeline_config["evaluation"]["amp"] if "amp" in pipeline_config["evaluation"] else False
+
+        evaluations: dict[int, EvaluationStatusTracker] = {}
+
+        # enforce unique dataset_ids
+        dataset_ids_seen = set()
```

Comment:
I think you could also do something like
```
dataset_ids = [dataset["dataset_id"] for dataset in  pipeline_config["evaluation"]["datasets"]]
if len(set(dataset_ids)) < len(dataset_ids):
    raise some exception

```

to decouple this check from preparing the input to the grpc request (and avoid the indentation struggle below)

Reply:
I agree, done in supervisor.